cms/djangoapps/contentstore/rest_api/v0/views/advanced_settings.py

```diff
@@ -0,0 +1,152 @@
+                "value": "Studio",
+                "display_name": "Course Editor",
+                "help": "Enter the method by which this course is edited (\"XML\" or \"Studio\").",
+                "deprecated": true,
+                "hide_on_enabled_publisher": false
+            },
+            "days_early_for_beta": {
+                "value": null,
+                "display_name": "Days Early for Beta Users",
+                "help": "Enter the number of days before the start date that beta users can access the course.",
+                "deprecated": false,
+                "hide_on_enabled_publisher": false
+            },
+            ...
+        }
+        ```
+        """
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_read_access(request.user, course_key):
+            raise self.permission_denied(request)
+        course_module = modulestore().get_course(course_key)
+        return Response(CourseMetadata.fetch(course_module))
+
+    @apidocs.schema(
+        body=CourseAdvancedSettingsSerializer,
+        parameters=[apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID")],
+        responses={
+            200: CourseAdvancedSettingsSerializer,
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    @verify_course_exists
+    def patch(self, request: Request, course_id: str):
+        """
+        Update a course's advanced settings.
+
+        **Example Request**
+
+            PATCH /api/contentstore/v0/advanced_settings/{course_id} {
+                "{setting_name}": {
+                    "value": {setting_value}
+                }
+            }
+
+        **PATCH Parameters**
+
+        The data sent for a patch request should follow a similar format as
+        is returned by a ``GET`` request. Multiple settings can be updated in
+        a single request, however only the ``value`` field can be updated
+        any other fields, if included, will be ignored.
+
+        Here is an example request that updates the ``advanced_modules``
+        available for the course, and enables the calculator tool:
+
+        ```json
+        {
+            "advanced_modules": {
+                "value": [
+                    "poll",
+                    "survey",
+                    "drag-and-drop-v2",
+                    "lti_consumer"
+                ]
+            },
+            "show_calculator": {
+                "value": true
+            }
+        }
+        ```
+
+        **Response Values**
+
+        If the request is successful, an HTTP 200 "OK" response is returned,
+        along with all the course's settings similar to a ``GET`` request.
+        """
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_write_access(request.user, course_key):
```

Comment:
Can we move this to a method decorator as well? as this only applies to `patch` we can use a decorator for the `patch` method.

Reply:
Do you mean the permission check? They are different checks for get and post, so can't be moved to a common decorator.

Reply:
sorry, do you mean we can't have separate decorators for `get` and `post`?